tool/python/examples/train_cifar10.py

```diff
@@ -0,0 +1,136 @@
+#* under the License.
+#*
+#*************************************************************/
+
+import os, sys, string
+import numpy as np
+
+current_path_ = os.path.dirname(__file__)
+singa_root_=os.path.abspath(os.path.join(current_path_,'../../..'))
+sys.path.append(os.path.join(singa_root_,'tool','python'))
+
+from singa.driver import Driver
+from singa.layer import *
+from singa.model import *
+from singa.utils.utility import swap32
+from google.protobuf.text_format import Merge
+
+'''
+Example of CNN with cifar10 dataset
+'''
+
+fname_mean_image = 'tool/python/examples/datasets/cifar10_mean_image'
+mean_image = np.fromfile(fname_mean_image)
+
+def load_dataset(did=1):
+    '''
+       5 binary dataset, each contains 10000 images
+       1 row (1 image) includes 1 label & 3072 pixels
+       3072 pixels are  3 channels of a 32x32 image
+    '''
+    print '[Loading CIFAR10 dataset]', did
+    dataset_dir_ = singa_root_ + "/examples/cifar10/cifar-10-batches-bin"
+    fname_train_data = dataset_dir_ + "/data_batch_{}.bin".format(did)
+    
+    nb_samples = 10000
+    nb_pixels = 3 * 1024  
+    d = np.fromfile(fname_train_data, dtype=np.uint8)
+    d = d.reshape(nb_samples, nb_pixels + 1) # +1 for label
+    x = d[:, 1:] 
+    x = x - mean_image
+    print '   data x:', x.shape
+    y = d[:, 0]
+    y = y.reshape(nb_samples, 1) 
+    print '  label y:', y.shape
+    return x, y
+
+def get_labellist():
+    dataset_dir_ = singa_root_ + "/examples/cifar10/cifar-10-batches-bin"
+    fname_label_list = dataset_dir_ + "/batches.meta.txt"
+    label_list_ = np.genfromtxt(fname_label_list, dtype=str)
+    return label_list_
+
+#-------------------------------------------------------------------
+print '[Layer registration/declaration]'
+d = Driver()
+d.Init(sys.argv)
+
+input = Dummy()
+label = Dummy()
+
+conv1 = Convolution2D(32, 5, 1, 2, w_std=0.0001, b_lr=2)
+pool1 = MaxPooling2D(pool_size=(3,3), stride=2)
+act1 = Activation('relu')
+lrn1 = LRN2D(3, alpha=0.00005, beta=0.75)
+
+conv2 = Convolution2D(32, 5, 1, 2, b_lr=2)
+act2 = Activation('relu')
+pool2 = AvgPooling2D(pool_size=(3,3), stride=2)
+lrn2 = LRN2D(3, alpha=0.00005, beta=0.75)
+
+conv3 = Convolution2D(64, 5, 1, 2)
+act3 = Activation('relu')
+pool3 = AvgPooling2D(pool_size=(3,3), stride=2)
+
+inner1 = Dense(10, w_wd=250, b_lr=2, b_wd=0)
+#act4 = Activation('softmax')
+loss = Loss('softmaxloss')
+
+sgd = SGD(decay=0.004, momentum=0.9, lr_type='manual', step=(0,60000,65000), step_lr=(0.001,0.0001,0.00001))
+
+#-------------------------------------------------------------------
+print '[Start training]'
+batchsize = 100 
+disp_freq = 50
+train_step = 1000
+
+for dataset_id in range(train_step / batchsize):
+
+    x, y = load_dataset(dataset_id%5+1)
+
+    for i in range(x.shape[0] / batchsize):
+
+        xb, yb = x[i*batchsize:(i+1)*batchsize,:], y[i*batchsize:(i+1)*batchsize,:]
+        input.SetData(xb, 3, 0)
+        label.SetData(yb, 1, 1)
+        
+        conv1.ComputeFeature(input)
```

Comment:
possible to do
(i) if user create a model m (like what we do in singa-125), then we can traverse it 
or
(ii) create a list or dict for maintaining a network in [conv1, pool1, act1, ... ] and go through it.

I will work on it tomorrow. pls let me know if you have better suggestion.

Thanks.